### ISSUE TITLE
在Ade20K数据集语义分割问题

### ISSUE THREAD

**Issue by Destinycjk** [PaddlePaddle/PaddleSeg#2798]
My environment: 
 1. PaddleSeg version: PaddleSeg release/2.5
 2. PaddlePaddle version: PaddlePaddle 2.4.0.post112
 3. Operation system: Windows
 4. Python version: Python3.8
 5. CUDA/cuDNN version:  CUDA11.2/cuDNN 8.2

安装完PaddlePaddle环境以及paddleseg后，参考quick start能够正常运行。但是在测试configs/segmenter文件下segmenter_vit_base_mask_ade20k_512x512_160k.yml这个配置文件时，shell在运行一下后就自动结束了，下面是命令代码与完整日志：
(PaddleSeg_CUDA102) D:\BUAA_CK\CODE\PaddleSeg\PaddleSeg-release-2.5>python train.py  --config configs/segmenter/segmenter_vit_small_mask_ade20k_512x512_160k.yml --do_eval  --use_vdl --save_interval 5000 --save_dir output
D:\BUAA_CK\CODE\PaddleSeg\PaddleSeg-release-2.5\paddleseg\models\losses\decoupledsegnet_relax_boundary_loss.py:19: DeprecationWarning: Please use `shift` from the `scipy.ndimage` namespace, the `scipy.ndimage.interpolation` namespace is deprecated.
  from scipy.ndimage.interpolation import shift
D:\BUAA_CK\CODE\PaddleSeg\PaddleSeg-release-2.5\paddleseg\transforms\functional.py:18: DeprecationWarning: Please use `distance_transform_edt` from the `scipy.ndimage` namespace, the `scipy.ndimage.morphology` namespace is deprecated.
  from scipy.ndimage.morphology import distance_transform_edt
2022-12-01 15:13:55 [INFO]
------------Environment Information-------------
platform: Windows-10-10.0.19044-SP0
Python: 3.8.15 | packaged by conda-forge | (default, Nov 22 2022, 08:42:03) [MSC v.1929 64 bit (AMD64)]
Paddle compiled with cuda: True
NVCC: Build cuda_11.2.r11.2/compiler.29373293_0
cudnn: 8.2
GPUs used: 1
CUDA_VISIBLE_DEVICES: 0
GPU: ['GPU 0: NVIDIA GeForce']
PaddleSeg: 2.5.0
PaddlePaddle: 2.4.0
OpenCV: 4.6.0
------------------------------------------------
2022-12-01 15:13:55 [INFO]
---------------Config Information---------------
batch_size: 2
iters: 160000
loss:
  coef:
  - 1
  types:
  - ignore_index: 255
    type: CrossEntropyLoss
lr_scheduler:
  end_lr: 1.0e-05
  learning_rate: 0.001
  power: 0.9
  type: PolynomialDecay
model:
  backbone:
    depth: 12
    drop_path_rate: 0.1
    drop_rate: 0.0
    embed_dim: 384
    final_norm: true
    img_size: 512
    mlp_ratio: 4
    num_heads: 6
    patch_size: 16
    pretrained: https://bj.bcebos.com/paddleseg/dygraph/pretrained_models/vit_small_patch16_384_augreg.tar.gz
    qkv_bias: true
    type: VisionTransformer
  h_depth: 2
  h_drop_path_rate: 0.1
  h_drop_rate: 0.0
  h_embed_dim: 384
  h_mlp_ratio: 4
  h_num_heads: 6
  type: MaskSegmenter
optimizer:
  momentum: 0.9
  type: sgd
  weight_decay: 0.0
test_config:
  crop_size:
  - 512
  - 512
  is_slide: true
  stride:
  - 512
  - 512
train_dataset:
  dataset_root: data/ADEChallengeData2016/
  mode: train
  transforms:
  - max_scale_factor: 2.0
    min_scale_factor: 0.5
    scale_step_size: 0.25
    type: ResizeStepScaling
  - crop_size:
    - 512
    - 512
    type: RandomPaddingCrop
  - type: RandomHorizontalFlip
  - brightness_range: 0.4
    contrast_range: 0.4
    saturation_range: 0.4
    type: RandomDistort
  - type: Normalize
  type: ADE20K
val_dataset:
  dataset_root: data/ADEChallengeData2016/
  mode: val
  transforms:
  - short_size: 512
    type: ResizeByShort
  - type: Normalize
  type: ADE20K
------------------------------------------------
W1201 15:13:55.851404 12408 gpu_resources.cc:61] Please NOTE: device: 0, GPU Compute Capability: 5.2, Driver API Version: 11.8, Runtime API Version: 11.2
W1201 15:13:55.851404 12408 gpu_resources.cc:91] device: 0, cuDNN Version: 8.2.

然后就自动结束了，没有任何报错信息，没找到是哪里的问题，非常感谢您的回复！

**Comment by Bobholamovic** [comment]
在 Windows 系统中，若程序运行时内存耗尽，有可能出现这种没有提示就退出的情况。建议在执行程序的同时打开任务管理器观察内存用量。

**Comment by Destinycjk** [comment]
您好，我通过任务管理器监测了一下，如图所示。
![捕获](https://user-images.githubusercontent.com/46669305/205002124-965694a9-82c3-4ae2-b4f0-440c0f33b327.PNG)
可以看到在程序中断前，内存的状态是正常的，但是磁盘1的活动时间满了，GPU的也随之爆满了，请问这个问题该怎么解决呢？另外我把batchsize已经调整为1了，还是不行。

**Comment by Bobholamovic** [comment]
可以尝试减小图像的输入分辨率，具体操作为在配置文件中为`crop_size`项设较小的值。

**Comment by Destinycjk** [comment]
似乎不是这个原因，我把crop_size改成了128也不行，然后我用python predict.py --config configs/segmenter/segmenter_vit_base_mask_ade20k_512x512_160k.yml --model_path output/download/model.pdparams --image_path data/ADEChallengeData2016/images\validation/ADE_val_00000002.jpg --save_dir output/result 命令进行单张图片的预测，结果也同样中断了。理论上讲预测并不需要加载那么多图片吧。

**Comment by Destinycjk** [comment]
另外我用眼底分割的示例试了一下，也是没有问题的，然后我看到在我这边中断的下一步log显示的是加载预训练模型，如图所示。所以我这边中断的原因是不是在加载预训练模型上。
眼底分割的预训练模型是四十多mb，然后我想要训练的那个预训练模型是三百多mb，是不是在这里中断了呢？
![Uploading 捕获.PNG…]()

**Comment by Bobholamovic** [comment]
请检查预训练模型是否存在损坏的情况。可尝试将配置项中的`pretrained`参数设置为 `null`，观察是否问题依然存在。

**Comment by Destinycjk** [comment]
预训练模型为网络链接加载。我在将pretrained参数设置为null后，问题依然存在。。。日志如下，可以看到其中的pretrained参数已经设为null。
D:\BUAA_CK\CODE\PaddleSeg\PaddleSeg\paddleseg\models\losses\decoupledsegnet_relax_boundary_loss.py:19: DeprecationWarning: Please use `shift` from the `scipy.ndimage` namespace, the `scipy.ndimage.interpolation` namespace is deprecated.
  from scipy.ndimage.interpolation import shift
D:\BUAA_CK\CODE\PaddleSeg\PaddleSeg\paddleseg\transforms\functional.py:18: DeprecationWarning: Please use `distance_transform_edt` from the `scipy.ndimage` namespace, the `scipy.ndimage.morphology` namespace is deprecated.
  from scipy.ndimage.morphology import distance_transform_edt
2022-12-01 17:41:19 [INFO]
------------Environment Information-------------
platform: Windows-10-10.0.19044-SP0
Python: 3.8.15 | packaged by conda-forge | (default, Nov 22 2022, 08:42:03) [MSC v.1929 64 bit (AMD64)]
Paddle compiled with cuda: True
NVCC: Build cuda_11.2.r11.2/compiler.29373293_0
cudnn: 8.2
GPUs used: 1
CUDA_VISIBLE_DEVICES: 0
GPU: ['GPU 0: NVIDIA GeForce']
PaddleSeg: 2.5.0
PaddlePaddle: 2.4.0
OpenCV: 4.6.0
------------------------------------------------
2022-12-01 17:41:19 [INFO]
---------------Config Information---------------
batch_size: 2
iters: 160000
loss:
  coef:
  - 1
  types:
  - ignore_index: 255
    type: CrossEntropyLoss
lr_scheduler:
  end_lr: 1.0e-05
  learning_rate: 0.001
  power: 0.9
  type: PolynomialDecay
model:
  backbone:
    depth: 12
    drop_path_rate: 0.1
    drop_rate: 0.0
    embed_dim: 768
    final_norm: true
    img_size: 512
    mlp_ratio: 4
    num_heads: 12
    patch_size: 16
    pretrained: null
    qkv_bias: true
    type: VisionTransformer
  h_depth: 2
  h_drop_path_rate: 0.1
  h_drop_rate: 0.0
  h_embed_dim: 768
  h_mlp_ratio: 4
  h_num_heads: 12
  type: MaskSegmenter
optimizer:
  momentum: 0.9
  type: sgd
  weight_decay: 0.0
test_config:
  crop_size:
  - 512
  - 512
  is_slide: true
  stride:
  - 512
  - 512
train_dataset:
  dataset_root: data/ADEChallengeData2016/
  mode: train
  transforms:
  - max_scale_factor: 2.0
    min_scale_factor: 0.5
    scale_step_size: 0.25
    type: ResizeStepScaling
  - crop_size:
    - 512
    - 512
    type: RandomPaddingCrop
  - type: RandomHorizontalFlip
  - brightness_range: 0.4
    contrast_range: 0.4
    saturation_range: 0.4
    type: RandomDistort
  - type: Normalize
  type: ADE20K
val_dataset:
  dataset_root: data/ADEChallengeData2016/
  mode: val
  transforms:
  - short_size: 512
    type: ResizeByShort
  - type: Normalize
  type: ADE20K
------------------------------------------------
W1201 17:41:19.690460  5548 gpu_resources.cc:61] Please NOTE: device: 0, GPU Compute Capability: 5.2, Driver API Version: 11.8, Runtime API Version: 11.2
W1201 17:41:19.690460  5548 gpu_resources.cc:91] device: 0, cuDNN Version: 8.2.

**Comment by Bobholamovic** [comment]
这样的话问题有可能不是发生在模型加载权重的时候。建议添加断点或打 log，更准确地定位错误发生的位置，以便我们帮助你解决问题~

**Comment by Destinycjk** [comment]
好的，谢谢！我再研究一下。

**Comment by Destinycjk** [comment]
@Bobholamovic 您好，我后续在另一台电脑上，重新创建了几个conda的虚拟环境，cuda11.6和cuda11.2上进行了测试，均出现了同样的问题，即在运行到如下路时，程序自动中断。。。
![image](https://user-images.githubusercontent.com/46669305/205427332-1f446d52-e752-441f-b75e-b79224474376.png)

**Comment by Destinycjk** [comment]
已解决，显卡与CUDA、CUDNN的版本匹配问题。
